libdevice/cmake/modules/SYCLLibdevice.cmake

```diff
@@ -223,6 +235,58 @@ if (NOT MSVC AND UR_SANITIZER_INCLUDE_DIR)
     include/sanitizer_utils.hpp
     include/spir_global_var.hpp
     sycl-compiler)
+
+  set(sanitizer_generic_compile_opts ${compile_opts}
+                            -fno-sycl-instrument-device-code
+                            -I${UR_SANITIZER_INCLUDE_DIR})
+
+  set(asan_pvc_compile_opts_obj -fsycl -c
+                                ${sanitizer_generic_compile_opts}
+                                ${sycl_pvc_target_opt}
+                                -D__LIBDEVICE_PVC__)
+
+  set(asan_cpu_compile_opts_obj -fsycl -c
+                                ${sanitizer_generic_compile_opts}
+                                ${sycl_cpu_target_opt}
+                                -D__LIBDEVICE_CPU__)
+
+  set(asan_dg2_compile_opts_obj -fsycl -c
+                                ${sanitizer_generic_compile_opts}
+                                ${sycl_dg2_target_opt}
+                                -D__LIBDEVICE_DG2__)
+
+  set(asan_pvc_compile_opts_bc  ${bc_device_compile_opts}
```

Comment:
Hi, @mdtoguchi 
When building llvm .bc format device libraries, we use:
clang++ -fsycl -fsycl-device-only -fsycl-device-obj=llvmir xxx.cpp
Do we need to add -fsycl-targets=intel_gpu_pvc,... when building for AOT?
Thanks very much.

Reply:
`-fsycl-targets=intel_gpu_pvc` in general applies to the device linking step.  One gotcha here, is the device compilation when adding `-fsycl-targets=intel_gpu_pvc` when generating the device only IR, is the predefined macros that are specific to the PVC target.  If those are important the option will need to be added.

Reply:
For sanitizer device library impl, we don't rely on any compiler predefined macros that are specific to the PVC/DG2/CPU target, so removing the target options here should be OK.